### DISCUSSION TITLE
PDF-output: Placing the LoA into the `toc.tex`-template-partial

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [rchaput/acronyms#29] [Q&A]
Hello,

I have a bit of a peculiar issue:

I have a document which must render to both html and pdf. For HTML, I can just use 

``````qmd
::: {.content-visible when-format="html"}
\printacronyms

```{=latex}
\cleardoublepage
```
``````

within the document-body to put the LoA at the beginning of the doc, right before the first section 'Introduction'

In PDF, I currently use both a `before-body.tex`- and a `toc.tex`-template partial.

In order to place the the acronyms in the roman-numeral-numbered prefix section of the document, I would have to position it into the `toc.tex`-partial:

```tex

% toc.tex
$if(toc)$
\cleardoublepage
\pagenumbering{roman} % Start Roman page numbering here
\setcounter{page}{1}   % Start numbering from i
$if(toc-title)$
\renewcommand*\contentsname{$toc-title$}
$endif$
$if(colorlinks)$
\hypersetup{linkcolor=$if(toccolor)$$toccolor$$else$$endif$}
$endif$
\setcounter{tocdepth}{$toc-depth$}
\tableofcontents
\cleardoublepage
$endif$

$if(lot)$
\listoftables
\cleardoublepage
$endif$

$if(lof)$
\listoffigures
\cleardoublepage
$endif$

\printacronyms
\cleardoublepage

% Define and implement abstract
\newenvironment{abstract}{%
      \cleardoublepage
      \begin{center}
        \bfseries \abstractname
      \end{center}
      \begin{quote}
    }{%
      \end{quote}
    }
\begin{abstract}
$abstract$
\end{abstract}

```

Unfortunately, xelatex will try and recognise `\printacronyms` as a control-sequence, fail at doing so, and error out.

Is there any way to place the LoA into a tex-template-partial?

Thank you.
Sincerely,
~Gw

**Comment by rchaput** [comment]
Hi,

I do not know enough Quarto / LaTeX to give a definite answer, but here's what I think:

- `\printacronyms` must be parsed by Quarto/Pandoc to be replaced with the actual LoA content.
- These rendering do not happen within template partials, such as `toc.tex`.
- However, the templates contain variables that are substituted with their content, such as `$abstract$`, `$title$`, ...
- These variables may contain markup content (abstract may contain emphasis, italicized words, ...), and are taken from the document's metadata.

I believe the solution would be to change the metadata, in order to append a new variable containing the LoA, and to use this variable in the `toc.tex` template.
I wonder if simply using:

```yaml
---
title: "..."
loa: \printacronyms
---
```

would suffice, or if Quarto will ignore this string when in the metadata. If it works, it will be the easiest solution. Otherwise, we would need to change the metadata from Lua code; but I'm not sure whether the Lua code would be executed before the templates are actually rendered, and whether the metadata update from Lua would be visible during the template rendering... Additionally, because the `\printacronyms` would not be actually used, we would need some way to tell Lua to execute the command nonetheless.
